lib/blocks/blockreader.go

```diff
@@ -21,13 +23,55 @@ import (
 
 var lineWithLeadingSpacesMatcher = regexp.MustCompile("^[[:space:]]*(.*\n)$")
 
-type blockReadFunc func(*Reader, int, string) error
+type blockReadFunc func(*Reader, int, string, bool) error
 
 type BlockWriter interface {
 	Write(index, startLine, endLine int, text string)
 	Close() error
 }
 
+type blockFormat interface {
+	isStartingLine(line string) bool
+	isFinishLine(line string) bool
+	keepIndentation() bool
+}
+
+type markdownBlockFormat struct{}
+
+func (markdownBlockFormat) isStartingLine(line string) bool {
+	// nolint:gocritic
+	if strings.HasPrefix(line, "```hcl") { // documentation
+		return true
+	} else if strings.HasPrefix(line, "```terraform") { // documentation
+		return true
+	} else if strings.HasPrefix(line, "```tf") { // documentation
+		return true
+	}
+	return false
+}
+
+func (mbf markdownBlockFormat) isFinishLine(line string) bool {
+	return strings.HasPrefix(line, "```")
+}
+
+func (mbf markdownBlockFormat) keepIndentation() bool {
+	return false
+}
+
+type restructuredTextBlockFormat struct{}
```

Comment:
could we add some comments here to make it a little easier to reaD? as well update the names
```suggestion

// markdown based text
type markdownTextFormat struct{}

func (markdownBlockFormat) isStartingLine(line string) bool {
	// nolint:gocritic
	if strings.HasPrefix(line, "```hcl") { // documentation
		return true
	} else if strings.HasPrefix(line, "```terraform") { // documentation
		return true
	} else if strings.HasPrefix(line, "```tf") { // documentation
		return true
	}
	return false
}

func (mbf markdownBlockFormat) isFinishLine(line string) bool {
	return strings.HasPrefix(line, "```")
}

func (mbf markdownBlockFormat) keepIndentation() bool {
	return false
}

// restructed text format
type restructuredTextFormat struct{}
```